### PR TITLE
Fix fetch current location to locate on map for adding place

### PR DESCRIPTION
**Pull request by cp-ishita-g**
Simulator Screen Recording - iPhone 15 Plus - 2024-12-27 at 15 01 07

## Summary by CodeRabbit

- **New Features**
	- Introduced a new method in the Location Service to retrieve the current location based on user ID.

- **Bug Fixes**
	- Enhanced error handling for location fetching logic.

- **Refactor**
	- Updated location retrieval from a manager to a service, improving efficiency and performance.
	- Simplified equality and hash code implementations in state classes.

- **Documentation**
	- Added annotations for properties to ensure correct serialization.

### CODE REVIEW

File: app/lib/ui/flow/geofence/add/locate/locate_on_map_view_model.dart [hunk 1]
```diff
@@ -74,14 +84,15 @@ class LocateOnMapVieNotifier extends StateNotifier<LocateOnMapState> {
...
-      await placesService.addPlace(
-        spaceId,
-        placeName,
-        state.cameraLatLng!.latitude,
-        state.cameraLatLng!.longitude,
-        _currentUser!.id,
-        memberIds
-      );
+      if (_currentUser != null && state.cameraLatLng != null) {
+        await placesService.addPlace(
```
Comment (cp-radhika-s) [inline app/lib/ui/flow/geofence/add/locate/locate_on_map_view_model.dart:88]: It'll keep shoding progress on add button if position is null

Comment (cp-ishita-g) [reply app/lib/ui/flow/geofence/add/locate/locate_on_map_view_model.dart:88]: if position will null than it won't enable next button and there will be no further process

Comment (cp-radhika-s) [reply app/lib/ui/flow/geofence/add/locate/locate_on_map_view_model.dart:88]: Okay. Can you move `if (_currentUser != null && state.cameraLatLng != null)` to the top and return before changing any states?

File: data/lib/service/location_service.dart [hunk 2]
```diff
@@ -42,6 +42,19 @@ class LocationService {
     });
   }
 
+  Future<List<ApiLocation>?> getCurrentLocation(String userId) async {
```
Comment (cp-radhika-s) [inline data/lib/service/location_service.dart:45]: We only need one location so we can return latest location instead of list of single element

**Review by cp-radhika-s — CHANGES_REQUESTED** [review]

**Review by cp-radhika-s — APPROVED** [review]

**Pull request merged**